src/identity-providers/routes/IdentityProvider.ts

```diff
@@ -1,43 +1,26 @@
 import type { LocationDescriptorObject } from "history";
 import { generatePath } from "react-router-dom";
 import type { RouteDef } from "../../route-config";
-import { AddIdentityProvider } from "../add/AddIdentityProvider";
 import { DetailSettings } from "../add/DetailSettings";
 
 type IdentityProviderTabs = "settings" | "mappers";
 
-export type IdentityProviderCreateParams = {
+export type IdentityProviderParams = {
   realm: string;
   providerId?: string;
-};
-
-export type IdentityProviderParams = IdentityProviderCreateParams & {
   alias: string;
   tab: IdentityProviderTabs;
 };
 
 export const IdentityProviderRoute: RouteDef = {
-  path: "/:realm/identity-providers/:providerId/:alias/:tab",
+  path: "/:realm/identity-providers/:providerId?/:alias/:tab",
```

Comment:
Once `providerId` is made required it should be made required here as well. 
```suggestion
  path: "/:realm/identity-providers/:providerId/:alias/:tab",
```